lib/rubocop/cop/style/redundant_condition.rb

```diff
@@ -146,6 +172,16 @@ def synonymous_condition_and_branch?(node)
             !use_hash_key_access?(if_branch)
         end
 
+        def if_branch_is_true_type_and_else_is_not?(node)
+          return false unless node.ternary? || node.if?
+
+          _condition, if_branch, else_branch = *node # rubocop:disable InternalAffairs/NodeDestructuring
+
+          return false unless if_branch && else_branch
+
+          if_branch.true_type? && !else_branch.true_type?
```

Comment:
It can likely be simplified as follows.

```suggestion
          node.if_branch&.true_type? && !node.else_branch&.true_type?
```